server/sonar-server/src/main/java/org/sonar/server/activity/ActivityService.java

```diff
@@ -45,8 +46,15 @@ public void save(Activity activity) {
       .setAction(activity.getAction())
       .setMessage(activity.getMessage())
       .setData(KeyValueFormat.format(activity.getData()))
+      .setProfileKey((String) activity.getData().get("profileKey"))
```

Comment:
Why do you get the profileKey from the data of activity ? Why haven't you add a new property in Activity ?